src/App.js

```diff
@@ -43,19 +36,18 @@ function App() {
         </form>
 
         <ul className="foodList--ul">
-          {searchTerm.length > 0 ? (
-            filteredFood.length === 0 ? (
-              <p>Sorry, nothing matches that search</p>
-            ) : (
-              filteredFood.map((food) => {
-                return <FoodCard food={food} key={food.id}></FoodCard>;
-              })
-            )
-          ) : (
-            food.map((food) => {
-              return <FoodCard food={food} key={food.id}></FoodCard>;
-            })
-          )}
+          {food.map((item) => {
+            if (searchTerm) {
+              if (
+                validSearch(item.name, searchTerm) ||
+                validSearch(item.origin, searchTerm)
+              ) {
+                return <FoodCard key={item.id} food={item} />;
+              }
+            } else {
+              return <FoodCard key={item.id} food={item} />;
+            }
```

Comment:
Nailed it! You've simplified your state management by 33.3...%! That said (copy & pasting from my review of [Khalids PR](https://github.com/KhalidStonehouse2001/Projects/pull/1/files#diff-f24b941899e46a7e4a6382f77dcc25c2677bfd2451b0856397b7ae505de1a9e6R38-R43))

This is quite a cheeky condition because it implicitly means the map's predicate doesn't return anything if there's a search term but it doesn't match a particular item. More explicit code is generally more readable code.

My challenge to you is to find an array method that you can call on food before map that will remove the items that don't match your search term before you render them

Reply:
Khalid and I had similar thoughts on this refactor too, I'm not sure if I've used filter in the best way though